src/providers/twitch/TwitchChannel.cpp

```diff
@@ -437,6 +438,37 @@ void TwitchChannel::setLive(bool newLiveStatus)
     }
 }
 
+void TwitchChannel::refreshTitle(){
```

Comment:
This function is called from way too many places. It should probably only be called in the roomIdChanged signal, and when the user hovers over the header. And for the "on user hover over header" there needs to be throttling implemented, so the API call can't be spammed over and over again for the same streamer.